contracts/base/InterestConsensus.sol

```diff
@@ -0,0 +1,150 @@
+        ZeroCollateralCommon.Signature calldata signature,
+        address lender,
+        uint256 blockNumber,
+        uint256 interest
+    ) external onlySigner() isInitialized() {
+        require(
+            !hasSubmitted[msg.sender][lender][blockNumber],
+            "SIGNER_ALREADY_SUBMITTED"
+        );
+        hasSubmitted[msg.sender][lender][blockNumber] = true;
+
+        require(
+            lenderInfo.requestedInterestUpdate(lender) == blockNumber,
+            "INTEREST_NOT_REQUESTED"
+        );
+
+        require(
+            !nodeSubmissions[lender][blockNumber].finalised,
+            "INTEREST_ALREADY_FINALISED"
+        );
+
+        bytes32 hashedData = _hashData(
+            lender,
+            blockNumber,
+            interest,
+            signature.signerNonce
+        );
+        require(_signatureValid(signature, hashedData), "SIGNATURE_NOT_VALID");
+
+
+            ZeroCollateralCommon.AggregatedInterest memory aggregatedData
+         = nodeSubmissions[lender][blockNumber];
+
+        // if this is the first submission for this request
+        if (aggregatedData.totalSubmissions == 0) {
+            aggregatedData = ZeroCollateralCommon.AggregatedInterest({
+                totalSubmissions: 1,
+                minValue: interest,
+                maxValue: interest,
+                sumOfValues: interest,
+                finalised: false
+            });
+        } else {
+            if (interest < aggregatedData.minValue) {
+                aggregatedData.minValue = interest;
+            }
+            if (interest > aggregatedData.maxValue) {
+                aggregatedData.maxValue = interest;
+            }
+
+            aggregatedData.sumOfValues = aggregatedData.sumOfValues.add(interest);
+            aggregatedData.totalSubmissions++;
+        }
+
+        emit InterestSubmitted(msg.sender, lender, blockNumber, interest);
+
+        if (aggregatedData.totalSubmissions > requiredSubmissions) {
+            aggregatedData.finalised = true;
+
+            // average the submissions
+            uint256 finalInterest = aggregatedData.sumOfValues.div(
+                aggregatedData.totalSubmissions
+            );
+
+            require(
+                _resultsWithinTolerance(
+                    aggregatedData.maxValue,
+                    aggregatedData.minValue,
+                    finalInterest
+                ),
+                "MAXIMUM_TOLERANCE_SURPASSED"
+            );
+
+            lenderInfo.setAccruedInterest(lender, blockNumber, finalInterest);
+
+            emit InterestAccepted(lender, blockNumber, finalInterest);
+        }
+
+        nodeSubmissions[lender][blockNumber] = aggregatedData;
+    }
+
+    function _hashData(
+        address lender,
+        uint256 blockNumber,
+        uint256 interest,
+        uint256 signerNonce
+    ) internal pure returns (bytes32) {
+        return keccak256(abi.encode(lender, blockNumber, interest, signerNonce));
```

Comment:
Have you considered to add address(this) and/or Lenders contract address?

Reply:
Done!